### PR TITLE
M table generator new (from master)

### PR DESCRIPTION
**Pull request by tsuehr**
-Created a new branch from master
-Implemented JUnit Tests for MTableGenerator

### CODE REVIEW

**Comment by purbon** [comment]
I would recommend you to check the tests, see

```
:forbiddenApisMain
:forbiddenApisTestForbidden annotation use: org.junit.Test [defaultMessage Just name your test method testFooBar]
  in com.purbon.search.fair.MTableGeneratorTest (MTableGeneratorTest.java, annotation on method declaration of 'computeMTableWithValidParametersTest()')
Forbidden annotation use: org.junit.Test [defaultMessage Just name your test method testFooBar]
  in com.purbon.search.fair.MTableGeneratorTest (MTableGeneratorTest.java, annotation on method declaration of 'initializeWithInvalidKValueTest()')
Forbidden annotation use: org.junit.Test [defaultMessage Just name your test method testFooBar]
  in com.purbon.search.fair.MTableGeneratorTest (MTableGeneratorTest.java, annotation on method declaration of 'initializeWithInvalidNValueTest()')
Forbidden annotation use: org.junit.Test [defaultMessage Just name your test method testFooBar]
  in com.purbon.search.fair.MTableGeneratorTest (MTableGeneratorTest.java, annotation on method declaration of 'initializeWithInvalidPValueTest()')
Forbidden annotation use: org.junit.Test [defaultMessage Just name your test method testFooBar]
  in com.purbon.search.fair.MTableGeneratorTest (MTableGeneratorTest.java, annotation on method declaration of 'initializeWithInvalidAlphaValueTest()')
Scanned 3 class file(s) for forbidden API invocations (in 0.08s), 5 error(s).
 FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':forbiddenApisTest'.
> de.thetaphi.forbiddenapis.ForbiddenApiException: Check for forbidden API calls failed, see log.
```

elasticsearch uses as a styleguide the forbiddenapis (https://github.com/policeman-tools/forbidden-apis/wiki) to assure good code practices. I think error is straightforward, but if you get questions don't hesitate to ask 😄 good work!

**Comment by purbon** [comment]
@tsuehr I created https://github.com/tsuehr/fairsearch-elasticsearch-plugin/pull/1 to amend the current complaints in your test. If you merge them we can move forward and merge this pr 😄 

good work!

File: src/main/java/com/purbon/search/fair/utils/MTableGenerator.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        for (int i = 1; i < this.n + 1; i++) {
+            table[i] = m(i);
+        }
+        return table;
+    }
+
+    private Integer m(int k) {
+
+        BinomialDistribution dist = new BinomialDistribution(k, p);
+        return dist.inverseCumulativeProbability(alpha);
```
Comment (purbon) [inline src/main/java/com/purbon/search/fair/utils/MTableGenerator.java]: uau! as you said, code is not that different from what I had in my first try, isn't?

Comment (tsuehr) [reply src/main/java/com/purbon/search/fair/utils/MTableGenerator.java]: Yes^^ i think the method is pretty much the same what you wrote. I wasn't sure if you used the right function inverse Prob function but yes you did :D

**Comment by tsuehr** [comment]
I will look at the style guide

**Comment by purbon** [comment]
LGTM, :shipit: 

good work, next step the alpha correction function 😸

**Pull request merged**